hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/GenesisRecordsConsensusHook.java

```diff
@@ -68,7 +72,10 @@ public void process(@NonNull final TokenContext context) {
         final var blockStore = context.readableStore(ReadableBlockRecordStore.class);
 
         // This process should only run ONCE, when a node receives its first transaction after startup
-        if (blockStore.getLastBlockInfo().consTimeOfLastHandledTxn() != null) return;
+        //        if (blockStore.getLastBlockInfo().consTimeOfLastHandledTxn() != null) return;
+
+        // Not sure about this condition
+        if (blockStore.getLastBlockInfo().migrationRecordsStreamed()) return;
```

Comment:
I _think_ this line shouldn't be necessary. If we ever get past the check for `consTimeOfLastHandledTxn() != null)` then I believe this flag should always be false.

Reply:
@mhess-swl 
- we never get past the check for `consTimeOfLastHandledTxn() != null`
- a condition that I added works as expected -- synthetic records being published only once after starting the node (but I think there should be a better way)